### PR TITLE
Courses by season

### LINKED ISSUE #9: Limit Courses to specific Semester "season"
Some courses can only be taken during the Fall, or Spring semesters. Allow support so this can be specified in the course instance. If the semester is a fall semester and the course is only given in Spring, the course should not be added to that semester. The same should happen vice-versa. To obtain if the semester is a spring or fall semester, the modulus operand can be used.

### PR DESCRIPTION
**Pull request by Joshuacruzc**
Algorithm can now accommodate courses that are only given on spring or fall semesters.

Fixes #9

### CODE REVIEW

File: StudentPlan.py [hunk 1]
```diff
@@ -5,9 +5,18 @@ class StudentPlan:
...
 
-    def __init__(self, curriculum, max_credits):
+    def __init__(self, curriculum, max_credits, past_semesters=None):
         self.curriculum = curriculum
         self.max_credits = max_credits
+        if past_semesters is not None:
+            for semester_position in range(max(past_semesters.keys())+1):
+                past_semester = self.add_semester(semester_position, past=True)
+                try:
+                    courses_to_add = past_semesters[semester_position]
```
Comment (Everod52) [inline StudentPlan.py]: You can use the dictionary method get() and specify a default value, as a second parameter, if semester_position doesn't exist in past_semester. So you wouldn't have to use the try/except block.

Also, you can just check if the the key semester_position is in the dictionary past_semester.

File: StudentPlan.py [hunk 2]
```diff
@@ -23,50 +32,97 @@ def accommodate(self, course, correq=False):
...
+        self.past = past
+
+    def course_valid(self, course):
+        if not self.is_full and not self.past and self.credit_hours + course.get_credit_hours() <= self.max_credits:
+            if course.season == 2:
+                return self.position % 2 == 0
+            elif course.season == 1:
+                return self.position % 2 == 1
+            else:
+                return True
```
Comment (Everod52) [inline StudentPlan.py:80]: Why return True if course.season isn't 1 or 2? Is there another possible value for course.season?

File: StudentPlan.py [hunk 3]
```diff
@@ -23,50 +32,97 @@ def accommodate(self, course, correq=False):
...
         self.courses.append(course)
-        self.credit_hours += course.credit_hours
+        if course.lab:
+            course.lab.position = self.position
+            self.courses.append(course.lab)
+        self.credit_hours += course.get_credit_hours()
         self.is_full = self.credit_hours >= self.max_credits
 
+    def get_year(self):
+        return int(self.position/2) if self.position % 2 == 0 else self.position//2 + 1
```
Comment (Everod52) [inline StudentPlan.py:92]: int(self.position/2) --> self.position//2

**Review by Everod52 — CHANGES_REQUESTED** [review]
Minor changes suggested. Overall great job.

File: StudentPlan.py (hunk 3 above)
Comment (Joshuacruzc) [reply StudentPlan.py:92]: this is to eliminate the .0 at the end of the number.

File: StudentPlan.py (hunk 2 above)
Comment (Joshuacruzc) [reply StudentPlan.py:80]: Season is only defined when the course can only be taken in a first or second semester, if no such attribute is defined, then it should return true regardless if it is one season or the other.

File: StudentPlan.py (hunk 1 above)
Comment (Joshuacruzc) [reply StudentPlan.py]: Done

**Review by Everod52 — APPROVED** [review]

**Pull request merged**